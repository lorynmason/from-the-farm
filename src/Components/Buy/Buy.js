import React from 'react';
import Map from '../Map/Map';
import ProductList from '../ProductList/ProductList';
import Search from '../Search/Search';
import { connect } from 'react-redux';

export const Buy = (props) => {
  console.log(props);
  return (
    <section className="buy">
      <Search />
      <Map />
      <ProductList products={props.products} />
    </section>
  );
}

export const mapStateToProps = (state) => ({
  products: state.products
});

export default connect(mapStateToProps)(Buy);