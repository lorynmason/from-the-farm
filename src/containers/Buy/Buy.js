import React from 'react';
import Map from '../Map/Map';
import { ProductList } from '../../components/ProductList/ProductList';
import Search from '../Search/Search';
import { connect } from 'react-redux';
import { Loading } from '../../components/Loading/Loading';
import PropTypes from 'prop-types';

export const Buy = ({ productSearchResults, products, isLoading, history}) => {
 let productsToShow = products
 if(isLoading) {
   return (<Loading />)
 }
 if(productSearchResults.length) {
   productsToShow = products.filter( product => {
     return productSearchResults.includes(product.id)
   })
 }
  return (
    <section className="buy">
      <Search />
      <Map history={history}/>
      <ProductList products={productsToShow} />
    </section>
  );
}

export const mapStateToProps = (state) => ({
  products: state.products,
  productSearchResults: state.productSearchResults,
  isLoading: state.isLoading
});

export default connect(mapStateToProps)(Buy);

Buy.propTypes = {
  products: PropTypes.array,
  productSearchResults: PropTypes.array,
  isLoading: PropTypes.bool
}
