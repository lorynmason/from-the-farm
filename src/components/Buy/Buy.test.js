import React from 'react';
import { Buy } from './Buy';
import { shallow } from 'enzyme'

describe('Buy', () => {
  it('should match snapshot', () => {
    const state = {
      vendors : [],
      products: []
    }
    let wrapper = shallow(<Buy appState={state}/>)
    expect(wrapper).toMatchSnapshot()
  });
})
