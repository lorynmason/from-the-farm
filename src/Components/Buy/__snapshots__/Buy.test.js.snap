// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buy should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Buy
    appState={
      Object {
        "products": Array [],
        "vendors": Array [],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Search
          products={Array []}
        />,
        <Map
          vendors={Array []}
        />,
        <ProductList
          products={Array []}
        />,
      ],
      "className": "buy",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "products": Array [],
          "search": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "history": undefined,
          "vendors": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "products": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Search
            products={Array []}
          />,
          <Map
            vendors={Array []}
          />,
          <ProductList
            products={Array []}
          />,
        ],
        "className": "buy",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "products": Array [],
            "search": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "history": undefined,
            "vendors": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "products": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
